lib/task.js

```diff
@@ -183,13 +183,14 @@ export default class Task {
   */
   async dockerConfig(linkInfo) {
     let config = this.task.payload;
-    // Allow task specific environment vars to overwrite those provided by hooks
-    let env = _.defaults({}, config.env || {}, linkInfo.env || {});
 
     await this.runtime.privateKey.decryptEnvVariables(
       this.task.payload, this.status.taskId
```

Comment:
instead of giving `this.task.payload` can we give it `config` so that it's less confusing...

It's the same object I see, but there is a lot of room of doing it wrong right here :)